Meta/Keywords/scripting_library/TA/trigger/eval_triggered.py

```diff
@@ -0,0 +1,244 @@
+        tentacle_class,
+        time_frame=None,
+        symbol=None,
+        trigger=False,
+        value_key=commons_enums.CacheDatabaseColumns.VALUE.value,
+        cache_key=None,
+        config_name=None,
+        config: dict = None
+):
+    tentacle_class = tentacles_manager_api.get_tentacle_class_from_string(tentacle_class) \
+        if isinstance(tentacle_class, str) else tentacle_class
+    config_name = context.get_config_name_or_default(tentacle_class, config_name)
+    init_key = _get_init_key(context, config_name)
+    is_first_candle_only = _is_first_candle_only(context)
+    should_trigger = not is_first_candle_only or (is_first_candle_only and _is_first_candle_call(context, init_key))
+    if not context.symbol_writer.are_data_initialized_by_key.get(init_key, False) or (should_trigger and trigger):
+        with context.adapted_trigger_timestamp(tentacle_class, config_name):
+            # always trigger when asked to then return the triggered evaluation return
+            return (await _trigger_single_evaluation(context, tentacle_class, value_key, cache_key,
+                                                     config_name, config, init_key))[0]
+    if tentacle_class.use_cache():
+        # try reading from cache
+        try:
+            with context.adapted_trigger_timestamp(tentacle_class, config_name):
+                await context.ensure_tentacle_cache_requirements(tentacle_class, config_name)
+                value, is_missing = await context.get_cached_value(value_key=value_key,
+                                                                   cache_key=cache_key,
+                                                                   tentacle_name=tentacle_class.__name__,
+                                                                   config_name=config_name)
+                if not is_missing:
+                    return value
+        except commons_errors.UninitializedCache as e:
+            if tentacle_class is not None and trigger is False:
+                raise commons_errors.UninitializedCache(f"Can't read cache from {tentacle_class} before initializing "
+                                                        f"it. Either activate this tentacle or set the 'trigger' "
+                                                        f"parameter to True (error: {e})") from None
+
+    _ensure_cache_when_set_value_key(value_key, tentacle_class)
+    # read from evaluation matrix
+    for value in _tentacle_values(context, tentacle_class, time_frame=time_frame, symbol=symbol):
+        return value
+
+
+async def evaluator_get_results(
+        context,
+        tentacle_class,
+        time_frame=None,
+        symbol=None,
+        trigger=False,
+        value_key=commons_enums.CacheDatabaseColumns.VALUE.value,
+        cache_key=None,
+        limit=-1,
+        max_history=False,
+        config_name=None,
+        config: dict = None
+):
+    cache_key = 10000000000.0 if max_history else cache_key
```

Comment:
What is key `10000000000.0`? I don't understand

Reply:
I changed it